### PR TITLE
Add mount.

### PR DESCRIPTION
**Pull request by Mjollnirs**
Add sys/mount.h
Has been tested.

```C#
class Program
{
    static unsafe void Main(string[] args)
    {
        string _overlay = "overlay";
        string _target = "/mnt/merged";
        string _option = "lowerdir=/mnt/lower3:/mnt/lower2:/mnt/lower1,upperdir=/mnt/upper,workdir=/mnt/work";

        Console.WriteLine("mount test!");
        fixed (byte* overlay = GetBytes(_overlay), target = GetBytes(_target), option = GetBytes(_option))
        {
            int rv = LibC.mount(overlay, target, overlay, LibC.MS_NODEV, option);
            Console.WriteLine(rv);
        }

        Console.ReadKey();
        Console.WriteLine("umount test!");
        fixed (byte* target = GetBytes(_target))
        {
            int rv = LibC.umount(target);
            Console.WriteLine(rv);
        }
    }

    static unsafe Span<byte> GetBytes(string str)
    {
        int byteLength = Encoding.UTF8.GetByteCount(str) + 1;
        Span<byte> bytes = new byte[byteLength];
        Encoding.UTF8.GetBytes(str, bytes);

        return bytes;
    }
}

### CODE REVIEW

**Comment by Mjollnirs** [comment]
PTAL @tmds

File: src/Sources/linux.common/mount.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        public static int MNT_FORCE = 1;
+        
+        public static int MNT_DETACH = 1;
+        
+        public static int MNT_EXPIRE = 1;
+        
+        public static int UMOUNT_NOFOLLOW = 1;
+        
+        [DllImport(libc, SetLastError = true)]
+        public static extern int mount(byte* source, byte* target, byte* filesystemtype, int mountflags, void* data);
```
Comment (tmds) [inline src/Sources/linux.common/mount.cs]: mountflags: change type to `ulong_t`

File: src/Sources/linux.common/mount.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        public static int MS_NOUSER = 1 << 31;
+
+        public static int MNT_FORCE = 1;
+        
+        public static int MNT_DETACH = 1;
+        
+        public static int MNT_EXPIRE = 1;
+        
+        public static int UMOUNT_NOFOLLOW = 1;
```
Comment (tmds) [inline src/Sources/linux.common/mount.cs]: Can you change these into properties, that is `=>` instead of `=`, and also use the `ulong_t` type.

**Comment by tmds** [comment]
Thanks! Please have a look at the comments.

File: src/Sources/linux.common/mount.cs (hunk 2 above)
Comment (Mjollnirs) [reply src/Sources/linux.common/mount.cs]: If use the `ulong_t` type, `MS_NOUSER` will get an error. Any ideas?

Comment (tmds) [reply src/Sources/linux.common/mount.cs]: The values are wrong.
```
_Static_assert(MS_NOUSER == 2147483648, "MS_NOUSER has unexpected value");
 ^
/tmp/2eac0d2b-6336-4378-8386-4b15817fcfc8.c:1435:1: error: static assertion failed: "MNT_DETACH has unexpected value"
 _Static_assert(MNT_DETACH == 1, "MNT_DETACH has unexpected value");
 ^
/tmp/2eac0d2b-6336-4378-8386-4b15817fcfc8.c:1438:1: error: static assertion failed: "MNT_EXPIRE has unexpected value"
 _Static_assert(MNT_EXPIRE == 1, "MNT_EXPIRE has unexpected value");
 ^
/tmp/2eac0d2b-6336-4378-8386-4b15817fcfc8.c:1441:1: error: static assertion failed: "UMOUNT_NOFOLLOW has unexpected value"
 _Static_assert(UMOUNT_NOFOLLOW == 1, "UMOUNT_NOFOLLOW has unexpected value");
 ^
```
UMOUNT_NOFOLLOW should be 8
MNT_EXPIRE should be 4
MNT_DETACH should be 2

MS_NOUSER is a challenge, glibc defines it as 1 << 31, while it should be 1U << 31.
You can add it to this lists of constants which get skipped:
https://github.com/tmds/Tmds.LibC/blob/6a270975be475b1f4dc55d70242314496408e29d/test/Tmds.LibC.Tests/ConstantsTests.cs#L49
Add a comment after it `// glibc defines it as 1 << 31, while it should be 1U << 31.`.

**Comment by Mjollnirs** [comment]
Updated, please take a look.

**Comment by tmds** [comment]
Thanks @Mjollnirs, sorry it took a while for me to pick this up.
I don't have a release plan for this library, would you like to have a new release on nuget.org that includes this?

**Comment by tmds** [comment]
@Mjollnirs this is in the 0.3.0 version which is now on nuget.org.

**Comment by Mjollnirs** [comment]
I noticed, thank you very much.

**Pull request merged**